### ISSUE TITLE
"Error adding block" caused by ConcurrentModificationException

### ISSUE THREAD

**Issue by mjomble** [tectonicus/tectonicus#129]
I'm getting a lot of these errors when rendering a 1.16.3 map with Tectonicus 2.28.6
Is the latest version of Java supported?

```
> java --version
java 15.0.1 2020-10-20
Java(TM) SE Runtime Environment (build 15.0.1+9-18)
Java HotSpot(TM) 64-Bit Server VM (build 15.0.1+9-18, mixed mode, sharing)
```

Hopefully relevant parts of the output:

```
Minecraft jar version: 1.16.3
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
 + + + + + + + + + + + + + + + + + + + + + + +
                  Tectonicus
   Version: 2.28.6
   Build 490
   Constructed on October 20 2020 at 0930
...
World version: 1.16.3
...
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by io.netty.util.internal.PlatformDependent0$1 (file:/D:/games/Tectonicus-2.28.6/lib/Tectonicus-2.28.6.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of io.netty.util.internal.PlatformDependent0$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
...
Error adding block: block/bedrock in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
java.util.ConcurrentModificationException: null
        at java.util.HashMap.computeIfAbsent(HashMap.java:1225) ~[?:?]
        at tectonicus.renderer.Geometry.getMesh(Unknown Source) ~[Tectonicus-2.28.6.jar:?]
        at tectonicus.rasteriser.MeshUtil.addVertices(Unknown Source) ~[Tectonicus-2.28.6.jar:?]
        at tectonicus.rasteriser.MeshUtil.addBlock(Unknown Source) ~[Tectonicus-2.28.6.jar:?]
        at tectonicus.blockTypes.BlockModel.createGeometry(Unknown Source) [Tectonicus-2.28.6.jar:?]
        at tectonicus.blockTypes.BlockStateModel.createGeometry(Unknown Source) [Tectonicus-2.28.6.jar:?]
        at tectonicus.Chunk.createGeometry(Unknown Source) [Tectonicus-2.28.6.jar:?]
        at tectonicus.world.World.draw(Unknown Source) [Tectonicus-2.28.6.jar:?]
        at tectonicus.world.World.draw(Unknown Source) [Tectonicus-2.28.6.jar:?]
        at tectonicus.TileRenderer.renderBaseTiles(Unknown Source) [Tectonicus-2.28.6.jar:?]
        at tectonicus.TileRenderer.output(Unknown Source) [Tectonicus-2.28.6.jar:?]
        at tectonicus.TectonicusApp.run(Unknown Source) [Tectonicus-2.28.6.jar:?]
        at tectonicus.TectonicusApp.main(Unknown Source) [Tectonicus-2.28.6.jar:?]
Error adding block: block/diorite in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/stone in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/coal_ore in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/iron_ore in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/dirt in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/grass_block in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/grass_block in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/grass in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/grass_block in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/granite in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/tall_grass_bottom in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/tall_grass_top in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/acacia_leaves in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Error adding block: block/acacia_log in [ChunkCoord (-6, -2) / <-6, -2> ] in r.-1.-1.mca
...
Cleaning up...
Writing player skin cache info (1 skin to write)
100%
Player skin cache written
Finished
```

I've also attached r.-1.-1.mca in zipped form: [r.-1.-1.zip](https://github.com/tectonicus/tectonicus/files/5447991/r.-1.-1.zip)

**Comment by mjomble** [comment]
Also, no html file is written to the output dir.

**Comment by skoeven** [comment]
So the ConcurrentModificationExceptions are caused because I'm incorrectly modifying a map inside of a call to computeIfAbsent.  It looks like some changes were made in Java 15 to throw that exception where it didn't before.  I'll get it fixed.

I think you can ignore the illegal reflective access warnings.  I'll need to replace an old library to get rid of them but it should still work for now.

**Comment by skoeven** [comment]
I went ahead and removed the library causing the illegal reflective access warnings as it wasn't used in as many places as I thought and was pretty quick to do.

I've released a new version: [https://github.com/tectonicus/tectonicus/releases/tag/v2.28.7](https://github.com/tectonicus/tectonicus/releases/tag/v2.28.7)

**Comment by mjomble** [comment]
Confirmed fix in 2.28.7, thanks!
